### PR TITLE
Support attaching Simmetrix MeshNex numberings to vertices

### PR DESCRIPTION
**Pull request by cwsmith**
This PR adds support for loading a Simmetrix MeshNex (`.nex`) file with the Simmetrix mesh and attaching the numbering associated with the vertices.  Two CTest tests for `meshsim2osh` were also added.

The `meshsim2osh` utility takes an optional argument for the path to the `.nex` file.  An optional `-reorder` flag was also added to support running `oshdiff` (it requires a global numbering to exist) as part of the added tests.

Only simplex meshes are currently supported.

### CODE REVIEW

**Comment by jacobmerson** [comment]
how is the next file generated?

**Comment by cwsmith** [comment]
> how is the nex file generated?

The `MeshNex` API in the `MeshSim` module of SimModSuite supports it.  The code I used to transfer the numbering for the d3d mesh is here: https://github.com/SCOREC/simExportMeshOfSubDomain.

**Comment by cwsmith** [comment]
> how is the nex file generated?

The `MeshNex` API in the `MeshSim` module of SimModSuite supports it.  The code I used to transfer the numbering generated by TOMMS for the d3d core mesh is here: https://github.com/SCOREC/simExportMeshOfSubDomain.

File: src/Omega_h_meshsim.cpp [hunk 1]
```diff
@@ -621,17 +647,30 @@ Mesh read(filesystem::path const& mesh_fname, filesystem::path const& mdl_fname,
   pProgress p = NULL;
   pGModel g = GM_load(mdl_fname.c_str(), nm, p);
   pMesh m = M_load(mesh_fname.c_str(), g, p);
+  const bool hasNumbering = (numbering_fname.native() != std::string(""));
+  pMeshNex numbering = hasNumbering ? MeshNex_load(numbering_fname.c_str(), m) : NULL;
```
Comment (jacobmerson) [inline src/Omega_h_meshsim.cpp:651]: nullptr

File: src/Omega_h_file.hpp [hunk 2]
```diff
@@ -33,6 +33,8 @@ void write_sol(Mesh* mesh, std::string const& filepath,
 namespace meshsim {
 Mesh read(filesystem::path const& mesh, filesystem::path const& model,
           CommPtr comm);
+Mesh read(filesystem::path const& mesh, filesystem::path const& model,
```
Comment (jacobmerson) [inline src/Omega_h_file.hpp:36]: Let's add documentation on this function. At a minimum it needs to say that the numbering is a `.nex` file.

File: src/Omega_h_meshsim.cpp [hunk 3]
```diff
@@ -611,8 +637,8 @@ void read_internal(pMesh m, Mesh* mesh) {
   return;
 }
 
-Mesh read(filesystem::path const& mesh_fname, filesystem::path const& mdl_fname,
-    CommPtr comm) {
+Mesh readImpl(filesystem::path const& mesh_fname, filesystem::path const& mdl_fname,
+    filesystem::path const& numbering_fname, CommPtr comm) {
```
[the remark this answers is not in the document]
Comment (cwsmith) [reply src/Omega_h_meshsim.cpp:641]: The `filesystem` use here is from the Omega_h namespace:

https://github.com/SCOREC/omega_h/blob/master/src/Omega_h_filesystem.hpp

File: src/Omega_h_meshsim.cpp (hunk 1 above)
Comment (cwsmith) [reply src/Omega_h_meshsim.cpp]: Thank you.  This is fixed here https://github.com/SCOREC/omega_h/pull/40/commits/3aed91a71a5122a2ebd8c84b1f9c4f0bc8d53433.

File: src/Omega_h_file.hpp (hunk 2 above)
Comment (cwsmith) [reply src/Omega_h_file.hpp:51]: Docs added with https://github.com/SCOREC/omega_h/pull/40/commits/1b89b5e560616b5ee7f09a875d78cf57324ca17c.

File: src/meshsim2osh.cpp [hunk 4]
```diff
@@ -12,12 +12,26 @@ int main(int argc, char** argv) {
...
+  std::string numbering_in;
+  if (cmdline.parsed("-numbering")) {
+    std::cout << "attaching numbering...\n";
+    numbering_in = cmdline.get<std::string>("-numbering", "numbering-in");
+  }
+  auto mesh = Omega_h::meshsim::read(mesh_in, model_in, numbering_in, comm);
   auto family = mesh.family();
+  if (cmdline.parsed("-reorder")) {
+    std::cout << "reordering...\n";
+    Omega_h::reorder_by_hilbert(&mesh);
```
[the remark this answers is not in the document]
Comment (cwsmith) [reply src/meshsim2osh.cpp]: https://github.com/SCOREC/omega_h/pull/40/commits/92cb1e80dbcaf5d23bed4e88a348ffa76548763d removes the `-reorder` flag and uses `osh_reorder` to produce meshes that can be compared with `osh_diff` for ctest.

**Pull request merged**